src/libs/actions/PersonalDetails.ts

```diff
@@ -1,97 +1,90 @@
-    const userDetails = personalDetail || lodashGet(allPersonalDetails, login);
+    const userDetails = personalDetail ?? allPersonalDetails?.[login];
 
     if (!userDetails) {
         return userLogin;
     }
 
-    const firstName = userDetails.firstName || '';
-    const lastName = userDetails.lastName || '';
+    const firstName = userDetails.firstName ?? '';
+    const lastName = userDetails.lastName ?? '';
     const fullName = `${firstName} ${lastName}`.trim();
 
     return fullName || userLogin;
 }
 
 /**
- * @param {String} userAccountIDOrLogin
- * @param {String} [defaultDisplayName] display name to use if user details don't exist in Onyx or if
+ * @param [defaultDisplayName] display name to use if user details don't exist in Onyx or if
  *                                      found details don't include the user's displayName or login
- * @returns {String}
  */
-function getDisplayNameForTypingIndicator(userAccountIDOrLogin, defaultDisplayName = '') {
+function getDisplayNameForTypingIndicator(userAccountIDOrLogin: string, defaultDisplayName = ''): string {
     // Try to convert to a number, which means we have an accountID
     const accountID = Number(userAccountIDOrLogin);
 
     // If the user is typing on OldDot, userAccountIDOrLogin will be a string (the user's login),
     // so Number(string) is NaN. Search for personalDetails by login to get the display name.
-    if (_.isNaN(accountID)) {
-        const detailsByLogin = _.findWhere(allPersonalDetails, {login: userAccountIDOrLogin}) || {};
-        return detailsByLogin.displayName || userAccountIDOrLogin;
+    if (Number.isNaN(accountID)) {
+        const detailsByLogin = Object.entries(allPersonalDetails ?? {}).find(([, value]) => value?.login === userAccountIDOrLogin)?.[1];
+        return detailsByLogin?.displayName ?? userAccountIDOrLogin;
     }
 
-    const detailsByAccountID = lodashGet(allPersonalDetails, accountID, {});
-    return detailsByAccountID.displayName || detailsByAccountID.login || defaultDisplayName;
+    const detailsByAccountID = allPersonalDetails?.[accountID];
+    return detailsByAccountID?.displayName ?? detailsByAccountID?.login ?? defaultDisplayName;
 }
 
 /**
  * Gets the first and last name from the user's personal details.
  * If the login is the same as the displayName, then they don't exist,
  * so we return empty strings instead.
- * @param {Object} personalDetail
- * @param {String} personalDetail.login
- * @param {String} personalDetail.displayName
- * @param {String} personalDetail.firstName
- * @param {String} personalDetail.lastName
- *
- * @returns {Object}
  */
-function extractFirstAndLastNameFromAvailableDetails({login, displayName, firstName, lastName}) {
-    if (firstName || lastName) {
-        return {firstName: firstName || '', lastName: lastName || ''};
+function extractFirstAndLastNameFromAvailableDetails({login, displayName, firstName, lastName}: PersonalDetails): FirstAndLastName {
+    if (firstName ?? lastName) {
```

Comment:
Same here. Please consider case when `firstName` is empty string and `lastName` exists